numpy/core/tests/test_scalar_methods.py

```diff
@@ -201,3 +202,227 @@ def test_bit_count(self):
+        MAXEXP = finfo.maxexp
+        NMANT = finfo.nmant
+
+        # two spellings of infinity, with optional signs; case-insensitive
+        self.identical(ftype.fromhex('inf'), np.inf)
+        self.identical(ftype.fromhex('+Inf'), np.inf)
+        self.identical(ftype.fromhex('-INF'), -np.inf)
+        self.identical(ftype.fromhex('iNf'), np.inf)
+        self.identical(ftype.fromhex('Infinity'), np.inf)
+        self.identical(ftype.fromhex('+INFINITY'), np.inf)
+        self.identical(ftype.fromhex('-infinity'), -np.inf)
+        self.identical(ftype.fromhex('-iNFiNitY'), -np.inf)
+
+        # nans with optional sign; case insensitive
+        self.identical(ftype.fromhex('nan'), np.nan)
+        self.identical(ftype.fromhex('+NaN'), np.nan)
+        self.identical(ftype.fromhex('-NaN'), np.nan)
+        self.identical(ftype.fromhex('-nAN'), np.nan)
+
+        # variations in input format
+        self.identical(ftype.fromhex('1'), 1.0)
+        self.identical(ftype.fromhex('+1'), 1.0)
+        self.identical(ftype.fromhex('1.'), 1.0)
+        self.identical(ftype.fromhex('1.0'), 1.0)
+        self.identical(ftype.fromhex('1.0p0'), 1.0)
+        self.identical(ftype.fromhex('01'), 1.0)
+        self.identical(ftype.fromhex('01.'), 1.0)
+        self.identical(ftype.fromhex('0x1'), 1.0)
+        self.identical(ftype.fromhex('0x1.'), 1.0)
+        self.identical(ftype.fromhex('0x1.0'), 1.0)
+        self.identical(ftype.fromhex('+0x1.0'), 1.0)
+        self.identical(ftype.fromhex('0x1p0'), 1.0)
+        self.identical(ftype.fromhex('0X1p0'), 1.0)
+        self.identical(ftype.fromhex('0X1P0'), 1.0)
+        self.identical(ftype.fromhex('0x1P0'), 1.0)
+        self.identical(ftype.fromhex('0x1.p0'), 1.0)
+        self.identical(ftype.fromhex('0x1.0p0'), 1.0)
+        self.identical(ftype.fromhex('+0x1p0'), 1.0)
+        self.identical(ftype.fromhex('0x01p0'), 1.0)
+        self.identical(ftype.fromhex('0x1p00'), 1.0)
+        self.identical(ftype.fromhex(' 0x1p0 '), 1.0)
+        self.identical(ftype.fromhex('\n 0x1p0'), 1.0)
+        self.identical(ftype.fromhex('0x1p0 \t'), 1.0)
+        self.identical(ftype.fromhex('0xap0'), 10.0)
+        self.identical(ftype.fromhex('0xAp0'), 10.0)
+        self.identical(ftype.fromhex('0xaP0'), 10.0)
+        self.identical(ftype.fromhex('0xAP0'), 10.0)
+        self.identical(ftype.fromhex('0xbep0'), 190.0)
+        self.identical(ftype.fromhex('0xBep0'), 190.0)
+        self.identical(ftype.fromhex('0xbEp0'), 190.0)
+        self.identical(ftype.fromhex('0XBE0P-4'), 190.0)
+        self.identical(ftype.fromhex('0xBEp0'), 190.0)
+        self.identical(ftype.fromhex('0xB.Ep4'), 190.0)
+        self.identical(ftype.fromhex('0x.BEp8'), 190.0)
+        self.identical(ftype.fromhex('0x.0BEp12'), 190.0)
+
+        # moving the point around
+
+        # results that should overflow...
+        large_values = [
+            f"-0x1p{MAXEXP}", f"0x1p+{MAXEXP + 1}", f"+0X1p{MAXEXP + 5}",
```

Comment:
Due to the fact we use `float32.fromhex` for `16`, the return value on underflow does not set an error when it overflows for `16` and simply leads to `(-)inf`. Need to see how we can handle this properly instead of checking at the caller again.

Reply:
resolved in 07b4614